### ISSUE TITLE
Inconsistent namespace behaviour using Terraform to read HCP Vault Secrets

### ISSUE THREAD

**Issue by tyu0912** [hashicorp/vault#21447]
**Describe the bug**

Received odd `No secrets found at...` error using Terraform to read secrets in HCP Vault 1.12.3+ent. Typically, we use the following block to read secrets and this has worked for us through numerous applications. 

```hcl
data "vault_generic_secret" "our_secret" {
  namespace = "admin/<child_namespace>"
  path      = "<secret_path>"
}
```

However, today we noticed that we had to remove the parent namespace, admin, in order for it to work. All other settings/configs seem to be consistent. Curious why/how this happened?

```hcl
data "vault_generic_secret" "our_secret" {
  namespace = "<child_namespace>"
  path      = "<secret_path>"
}
```

**To Reproduce**
- See above. 

**Expected behavior**
I'd expect the below block to work

```hcl
data "vault_generic_secret" "our_secret" {
  namespace = "admin/<child_namespace>"
  path      = "<secret_path>"
}
```

**Environment:**
HCP Vault - 1.12.3+ent
Terraform Vault Provider - 3.15.2
Terraform version - 1.5.1

Vault server configuration file(s):

The variables are specified in the Terraform Cloud workspace. 

```hcl
provider "vault" {
  auth_login_userpass {
    namespace = "admin"
    username  = var.hcp_vault_username
    password  = var.hcp_vault_password
  }
}
```

**Comment by maxb** [comment]
I have never actually used the `terraform-provider-vault` support for specifying namespaces within individual `data`/`resource` blocks - it came out too late for the project I was working on, so we did everything with `provider` level specification of namespaces. And now, I no longer work for a company with an Enterprise subscription.

However I do remember reading in <https://registry.terraform.io/providers/hashicorp/vault/latest/docs#namespace-support>:

> All resource's namespace will be made relative to the provider's configured namespace.

The behaviour you observe seems to suggest that your "provider's configured namespace" has changed from the root namespace to "admin".

The provider configuration snippet that you've shown does not set the "provider's configured namespace" (which is different from a namespace specified within any `auth` block), so it would pick up any namespace specified in a `VAULT_NAMESPACE` environment variable.

Is it possible that the `VAULT_NAMESPACE` environment variable has been set to `admin` in the context performing your Terraform runs?

**Comment by tyu0912** [comment]
Hi @maxb , I was able to resolve this. Turns out it was an obvious but easy-to-miss misspecification of the vault provider version. 

In version 3.15.2, you're able to specify parent and children namespaces together in the read but it seems in 3.16.0 they've updated it to be relative to whatever is specified in the auth block which I have as `admin` above. Not sure if this is a feature or bug but it does seem to have been reported elsewhere: https://github.com/hashicorp/terraform-provider-vault/issues/1903.
